client/session.go

```diff
@@ -1895,6 +1902,22 @@ func (b blockMetadatasByTime) Less(i, j int) bool {
 	return b[i].start.Before(b[j].start)
 }
 
+func newTimesByUnixNanos(values []int64) []time.Time {
+	result := make([]time.Time, len(values))
```

Comment:
Nit: Should we use nano uint64 as return type here and below to reduce mem usage?